test/integration/http_integration.cc

```diff
@@ -149,18 +149,36 @@ IntegrationCodecClient::startRequest(const Http::HeaderMap& headers) {
   return {encoder, std::move(response)};
 }
 
-void IntegrationCodecClient::waitForDisconnect() {
+bool IntegrationCodecClient::waitForDisconnect(uint32_t time_to_wait) {
+  Event::TimerPtr wait_timer;
+  bool wait_timer_triggered = false;
+  if (time_to_wait) {
+    wait_timer = connection_->dispatcher().createTimer([this, &wait_timer_triggered] {
+      connection_->dispatcher().exit();
+      wait_timer_triggered = true;
+    });
+    wait_timer->enableTimer(std::chrono::milliseconds(time_to_wait));
+  }
+
   connection_->dispatcher().run(Event::Dispatcher::RunType::Block);
+  if (wait_timer_triggered && !disconnected_) {
+    return false;
+  }
   EXPECT_TRUE(disconnected_);
```

Comment:
Should disable the timer if it hasn't fired.

Reply:
Done.